### PR TITLE
python: use os.login_tty() on pty stream channels

### LINKED ISSUE #19075: python: Terminal tab broken with fish shell
### Discussed in https://github.com/cockpit-project/cockpit/discussions/19067

<div type='discussions-op-text'>

<sup>Originally posted by **kulak** July  4, 2023</sup>
I've got this error message and I am not sure where to start:

```
warning: No TTY for interactive shell (tcgetpgrp failed)
setpgid: Inappropriate ioctl for device
disconnected
```

It is Arch Linux installation from <https://archlinux.org/packages/extra/x86_64/cockpit/></div>

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
When creating a pty-enabled stream channel we call setsid() but depend on the shell to take control of the terminal.  fish doesn't do that, leading to an error message, rendering the Terminal page unusable if the user has fish as their default shell.

Let's call `os.login_tty()` explicitly.

Fixes #19075

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
Reproducer from the issue:

```
$ cpf spawn fish pty=True : wait | spy -m cockpit.bridge
...
111
ch1
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
12
ch1
58
ch1
126
ch1
lis@toolbox ~/s/c/fishy-tty (fishy-tty)> 
```

looks a lot better.

**Comment by allisonkarlitskaya** [comment]
The part of this that we really need here is the `TIOCSCTTY` and Python 3.6 has that.  Let me try again.

**Review by jelly — APPROVED** [review]
Pulled the latest changes, seems to work nicely :+1:
